apps/ex_wire/lib/ex_wire/node_discovery_supervisor.ex

```diff
@@ -1,34 +1,37 @@
 defmodule ExWire.NodeDiscoverySupervisor do
-  use Supervisor
-
   @moduledoc """
   The Node Discovery Supervisor manages two processes. The first process is kademlia
   algorithm's routing table state - ExWire.Kademlia.Server,  the second one is process
-  that sends and receives all messages that are used for node discovery (ping, pong etc)
+  that sends and receives all messages that are used for node discovery (ping, pong etc).
   """
+  use Supervisor
 
   alias ExWire.{Config, Network}
   alias ExWire.Kademlia.Node
   alias ExWire.Kademlia.Server, as: KademliaServer
   alias ExWire.Struct.Endpoint
 
   def start_link(params \\ []) do
-    supervisor_name = discovery_param(params, :supervisor_name)
+    supervisor_name = Keyword.get(params, :supervisor_name, __MODULE__)
 
     Supervisor.start_link(__MODULE__, params, name: supervisor_name)
   end
 
   def init(params) do
-    {udp_module, udp_process_name} = discovery_param(params, :network_adapter)
-    kademlia_name = discovery_param(params, :kademlia_process_name)
-    port = discovery_param(params, :port)
-    bootnodes = (params[:nodes] || nodes()) |> Enum.map(&Node.new/1)
+    kademlia_name = Keyword.get(params, :kademlia_process_name, KademliaState)
```

Comment:
is there a reason this is configurable?

Reply:
Dunno, I just refactored to code to be clearer, but there's not an obvious reason why this process would be renamed by the caller.